astropy/time/core.py

```diff
@@ -282,6 +283,8 @@ def _get_time_fmt(self, val, val2, format, scale):
             try:
                 return FormatClass(val, val2, scale, self.precision,
                                    self.in_subfmt, self.out_subfmt)
+            except UnitConversionError:
```

Comment:
I could change that, but I don't think I'm comfortable with a bare except here, at least for now.  I agree it's a bit awkward though.